File: modules/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/config/HttpClientConfig.java
package com.roxiemobile.networkingapi.network.rest.config;

import java.util.List;

import okhttp3.Interceptor;

public interface HttpClientConfig
{
// MARK: - Methods

    /**
     * TODO
     */
    int connectTimeout();

    /**
     * TODO
     */
    int readTimeout();

    /**
     * TODO
     */
    List<Interceptor> interceptors();

    /**
     * TODO
     */
    List<Interceptor> networkInterceptors();
}

File: sources/modules/libraries/networking-api/src/main/java/com/roxiemobile/networkingapi/network/rest/configuration/HttpClientConfig.kt
package com.roxiemobile.networkingapi.network.rest.configuration

import com.roxiemobile.networkingapi.network.ssl.TlsConfig
import okhttp3.Interceptor

interface HttpClientConfig: Cloneable {

// MARK: - Properties

    /**
     * TODO
     */
    val requestTimeoutConfig: RequestTimeoutConfig?

    /**
     * TODO
     */
    val tlsConfig: TlsConfig?

    /**
     * TODO
     */
    val interceptors: List<Interceptor>?

    /**
     * TODO
     */
    val networkInterceptors: List<Interceptor>?

// MARK: - Methods

    /**
     * TODO
     */
    public override fun clone(): HttpClientConfig
}
